tests/e2e_logging.py

```diff
@@ -947,6 +947,40 @@ def id_for(i):
 
         infra.commit.wait_for_commit(c, seqno=last_seqno, view=view, timeout=3)
 
+        LOG.info("Checking error responses")
+        # Reversed range is illegal
+        r = c.get(
+            f"/app/log/public/historical/range?from_seqno={last_seqno}&to_seqno=1&id={id_a}"
+        )
+        assert r.status_code == http.HTTPStatus.BAD_REQUEST
+        assert r.body.json()["error"]["code"] == "InvalidInput"
+        r = c.get(
+            f"/app/log/public/historical/range?from_seqno={last_seqno}&to_seqno={last_seqno-1}&id={id_a}"
+        )
+        assert r.status_code == http.HTTPStatus.BAD_REQUEST
+        assert r.body.json()["error"]["code"] == "InvalidInput"
+
+        # Asking for future seqnos gives a clear error
+        # - First find latest valid seqno
+        r = c.get("/node/commit")
+        assert r.status_code == http.HTTPStatus.OK
+        last_valid_seqno = TxID.from_str(r.body.json()["transaction_id"]).seqno
+
+        # - Try a very invalid seqno
+        r = c.get(
+            f"/app/log/public/historical/range?to_seqno={last_valid_seqno*2}&id={id_a}"
+        )
+        assert r.status_code == http.HTTPStatus.BAD_REQUEST
+        assert r.body.json()["error"]["code"] == "InvalidInput"
+
+        # - Try the first invalid seqno
+        r = c.get(
+            f"/app/log/public/historical/range?to_seqno={last_valid_seqno+1}&id={id_a}"
```

Comment:
@maxtropets this is risky, if a new transaction takes place after the last_seqno on which we wait for commit at `:948`, for example a snapshot or a jwt refresh, you could easily get a 202 instead, which seems to be happening sometimes (#6417).

To be made robust, the test needs to be either more controlled to definitely rule this out (probably means a split env etc), or the invalid seqno needs to be a little more conservative (+10?).

Reply:
#6418